### PR TITLE
fix t.Fatal in a goroutine

### PR DESCRIPTION
**Pull request by chriscool**
This should fix at least part of issue #2043.

License: MIT
Signed-off-by: Christian Couder <chriscool@tuxfamily.org>

### CODE REVIEW

**Comment by chriscool** [comment]
Test failures are in Sharness tests and are not related.

File: test/supernode_client/main.go [hunk 1]
```diff
@@ -142,6 +142,7 @@ func sizeOfIthFile(i int64) int64 {
 }
 
 func runFileAddingWorker(n *core.IpfsNode) error {
+	errs := make(chan error, math.MaxInt32)
```
Comment (whyrusleeping) [inline test/supernode_client/main.go]: what... what is going on in this test? `MaxInt32` ?

your changes look good to me, but we should probably reevaluate what this test is doing...

**Comment by whyrusleeping** [comment]
Looks good! this may help flush out some issues, especially the ones in bitswap. Those might be the cause of our occasional bitswap test hang.

Only thing I would do is to not buffer the error channels, the pattern youre using should work fine with no buffering.

**Comment by chriscool** [comment]
@whyrusleeping ok thanks for your review.

I will wait a bit before for @jbenet or someone else's opinion before removing the error channel buffers though.

File: exchange/bitswap/bitswap_test.go [hunk 2]
```diff
@@ -168,19 +168,31 @@ func PerformDistributionTest(t *testing.T, numInstances, numBlocks int) {
 	t.Log("Distribute!")
 
 	wg := sync.WaitGroup{}
+	errs := make(chan error, 10)
```
Comment (jbenet) [inline exchange/bitswap/bitswap_test.go]: here this could be `make(chan error, len(instances))`

File: fuse/readonly/ipfs_test.go [hunk 3]
```diff
@@ -154,6 +155,7 @@ func TestIpfsStressRead(t *testing.T) {
 
 	// Now read a bunch, concurrently
 	wg := sync.WaitGroup{}
+	errs := make(chan error, 10)
```
Comment (jbenet) [inline fuse/readonly/ipfs_test.go]: `make(chan error, 4)`

File: merkledag/merkledag_test.go [hunk 4]
```diff
@@ -193,32 +194,43 @@ func runBatchFetchTest(t *testing.T, read io.Reader) {
 	}
 
 	wg := sync.WaitGroup{}
+	errs := make(chan error, 10)
```
Comment (jbenet) [inline merkledag/merkledag_test.go]: `len(dagservs)`

File: test/supernode_client/main.go (hunk 1 above)
Comment (jbenet) [inline test/supernode_client/main.go]: i believe this thing used to just sit, adding a new file every second. it could just as well loop forever.

Comment (jbenet) [inline test/supernode_client/main.go]: it's not really used.

**Comment by jbenet** [comment]
> Only thing I would do is to not buffer the error channels, the pattern youre using should work fine with no buffering.

yeah that sounds good to me, no buffering is fine

File: exchange/bitswap/bitswap_test.go (hunk 2 above)
Comment (jbenet) [inline exchange/bitswap/bitswap_test.go]: (or no buffering)

File: fuse/readonly/ipfs_test.go (hunk 3 above)
Comment (jbenet) [inline fuse/readonly/ipfs_test.go]: (or no buffering)

File: merkledag/merkledag_test.go (hunk 4 above)
Comment (jbenet) [inline merkledag/merkledag_test.go]: (or no buffering)

**Comment by jbenet** [comment]
this LGTM, other than no buffers

**Comment by chriscool** [comment]
Ok, I removed all the buffering.

**Pull request merged**